### PR TITLE
[Storage] IAM condition sample updates

### PR DESCRIPTION
**Pull request by frankyn**

### CODE REVIEW

**Comment by frankyn** [comment]
@TheRoyalTnetennba could you help review this PR?

File: storage/iam.rb [hunk 1]
```diff
@@ -62,14 +68,56 @@ def remove_bucket_iam_member project_id:, bucket_name:, role:, member:
...
-  bucket.policy do |policy|
-    policy.remove role, member
+  bucket.policy requested_policy_version: 3 do |policy|
+    policy.bindings.each do |binding|
+      if binding.role == role
+        if binding.members.count > 1
+          # At least 2 members exist in the binding
+          binding.members.delete member
+        else
+          # Last member in the list, delete binding from policy.
```
Comment (broady) [inline storage/iam.rb]: Is this necessary? Won't the server clean up this binding? What if it has a condition but no members? (Is that why you need to remove the binding?)

File: storage/iam.rb [hunk 2]
```diff
@@ -62,14 +68,56 @@ def remove_bucket_iam_member project_id:, bucket_name:, role:, member:
...
+          policy.bindings.remove binding
```
Comment (broady) [inline storage/iam.rb]: Is it valid to remove while iterating over the bindings list?

File: storage/iam.rb (hunk 1 above)
Comment (frankyn) [reply storage/iam.rb]: If there are no members in the array IAM will fail paraphrasing "at least 1 member is required".

File: storage/iam.rb (hunk 2 above)
Comment (frankyn) [reply storage/iam.rb]: :man_facepalming: good catch will fix.

File: storage/iam.rb (hunk 1 above)
Comment (frankyn) [reply storage/iam.rb]: Pending fix in the service this will go away. Until then the sample update is blocked.

**Comment by frankyn** [comment]
@GoogleCloudPlatform/yoshi-ruby anyone have a moment to review these changes from a Ruby perspective?

**Pull request merged**